# Top 25 Most Affordable Places to Raise a Family

Analysis by Alex Mahadevan  
Data journalist, [The Penny Hoarder]("https://www.thepennyhoarder.com")

### Data Cleaning

<p>We begin by importing the Pandas data analysis library.</p> 

In [1]:
import pandas as pd

<p>We used a combination of PANDAS and Excel to clean and merge all of the data herein. Sources include the [U.S. Census Bureau's American Community Survey](https://www.census.gov/acs/www/data/data-tables-and-tools/data-profiles/2015/), [University of Michigan's Institute for Social Research](http://www.icpsr.umich.edu/icpsrweb/NACJD/studies/35019), [The U.S. Bureau of Labor Statistics](https://data.bls.gov/map/MapToolServlet?survey=la&map=county&seasonal=u) and [The Robert Wood Johnson Foundation](http://www.countyhealthrankings.org/reports/2017-county-health-rankings-key-findings-report).
<p>The data cleaning process was daunting, dirty and rigorous, so we omitted it from the final analysis.</p>

<p>Now, we'll read in the initial dataset we'll be working from.</p>
<p>Notes:</p>
* The crime data for some smaller counties may be missing or underreported.
* We used counties instead of metropolitan statistical area or city data to have the most robust analysis with the widest number of variables, and to capture suburbs.
* Some of the data points were not available for Alaska and Hawaii. That's OK, since those states are pretty expensive to live in anyway (they'll likely be thrown out of the analysis).

In [2]:
df = pd.read_csv("/Users/alexmahadevan/Code/raise_a_family/county data.csv" , index_col=0)


<p>Here is a list of the variables we are considering.</p>

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Float64Index: 3220 entries, 1.0 to nan
Data columns (total 40 columns):
SUMLEV                        3141 non-null float64
REGION                        3141 non-null float64
DIVISION                      3141 non-null float64
STATE                         3141 non-null float64
COUNTY                        3141 non-null float64
STNAME                        3141 non-null object
CTY                           3141 non-null object
HEALTHCARE_COST               3134 non-null float64
PERCENT_FOOD_INSECURE         3135 non-null float64
CHILD_MORTALITY_RATE          1943 non-null float64
MENTAL_DISTRESS               3135 non-null float64
CHILDREN_UNINSURED            3135 non-null float64
DISCONNECTED_YOUTH            2046 non-null float64
DAILY_POLLUTION               3108 non-null float64
WATER_VIOLATION               3077 non-null float64
HOUSING_PROBLEMS              3135 non-null float64
POPESTIMATE2016               3141 non-null float64
AVG_NETM

<p>As you can see, for most variables, we have more than 3,100 counties to consider.</p>
<p>Since most people prefer to live where other people are living (and not in an igloo in Yakutat, Alaska), let's cull this list to only the counties with more population than the median.</p>
<p>First let's find the average.</p>

In [4]:
df.POPESTIMATE2016.mean()

102874.06080865966

<p>As you can see, the average population of a U.S. county as of 2016 was 102,874 (plus one twentieth of a person).</p>
<p>Let's drop the counties with a population less than that amount.</p>

In [5]:
df = df[df.POPESTIMATE2016 > 102874.06080865966]

In [6]:
df.COUNTY.describe()

count    584.000000
mean      90.041096
std      104.140746
min        1.000000
25%       25.000000
50%       67.000000
75%      113.000000
max      810.000000
Name: COUNTY, dtype: float64

<p>Now I'm going to tweak PANDAS so it doesn't revert every number to its scientific format</p>

In [7]:
pd.set_option('display.float_format', lambda x: '%.3f' % x)

In [40]:
df.fillna(0)

SUMLEV  REGION  DIVISION  STATE  COUNTY         STNAME  \
2.000     50.000   3.000     6.000  1.000   3.000        Alabama   
8.000     50.000   3.000     6.000  1.000  15.000        Alabama   
35.000    50.000   3.000     6.000  1.000  69.000        Alabama   
37.000    50.000   3.000     6.000  1.000  73.000        Alabama   
41.000    50.000   3.000     6.000  1.000  81.000        Alabama   
45.000    50.000   3.000     6.000  1.000  89.000        Alabama   
49.000    50.000   3.000     6.000  1.000  97.000        Alabama   
51.000    50.000   3.000     6.000  1.000 101.000        Alabama   
52.000    50.000   3.000     6.000  1.000 103.000        Alabama   
59.000    50.000   3.000     6.000  1.000 117.000        Alabama   
63.000    50.000   3.000     6.000  1.000 125.000        Alabama   
71.000    50.000   4.000     9.000  2.000  20.000         Alaska   
85.000    50.000   4.000     9.000  2.000 170.000         Alaska   
100.000   50.000   4.000     8.000  4.000   3.000        Arizona   
101.000   50.000   4.000     8.000  4.000   5.000        Arizona   
106.000   50.000   4.000     8.000  4.000  13.000        Arizona   
107.000   50.000   4.000     8.000  4.000  15.000        Arizona   
108.000   50.000   4.000     8.000  4.000  17.000        Arizona   
109.000   50.000   4.000     8.000  4.000  19.000        Arizona   
110.000   50.000   4.000     8.000  4.000  21.000        Arizona   
112.000   50.000   4.000     8.000  4.000  25.000        Arizona   
113.000   50.000   4.000     8.000  4.000  27.000        Arizona   
118.000   50.000   3.000     7.000  5.000   7.000       Arkansas   
130.000   50.000   3.000     7.000  5.000  31.000       Arkansas   
137.000   50.000   3.000     7.000  5.000  45.000       Arkansas   
174.000   50.000   3.000     7.000  5.000 119.000       Arkansas   
177.000   50.000   3.000     7.000  5.000 125.000       Arkansas   
180.000   50.000   3.000     7.000  5.000 131.000       Arkansas   
186.000   50.000   3.000     7.000  5.000 143.000       Arkansas   
191.000   50.000   4.000     9.000  6.000   1.000     California   
...          ...     ...       ...    ...     ...            ...   
3003.000  50.000   4.000     9.000 53.000   5.000     Washington   
3006.000  50.000   4.000     9.000 53.000  11.000     Washington   
3008.000  50.000   4.000     9.000 53.000  15.000     Washington   
3017.000  50.000   4.000     9.000 53.000  33.000     Washington   
3018.000  50.000   4.000     9.000 53.000  35.000     Washington   
3027.000  50.000   4.000     9.000 53.000  53.000     Washington   
3029.000  50.000   4.000     9.000 53.000  57.000     Washington   
3031.000  50.000   4.000     9.000 53.000  61.000     Washington   
3032.000  50.000   4.000     9.000 53.000  63.000     Washington   
3034.000  50.000   4.000     9.000 53.000  67.000     Washington   
3037.000  50.000   4.000     9.000 53.000  73.000     Washington   
3039.000  50.000   4.000     9.000 53.000  77.000     Washington   
3042.000  50.000   3.000     5.000 54.000   3.000  West Virginia   
3060.000  50.000   3.000     5.000 54.000  39.000  West Virginia   
3071.000  50.000   3.000     5.000 54.000  61.000  West Virginia   
3101.000  50.000   2.000     3.000 55.000   9.000      Wisconsin   
3109.000  50.000   2.000     3.000 55.000  25.000      Wisconsin   
3114.000  50.000   2.000     3.000 55.000  35.000      Wisconsin   
3126.000  50.000   2.000     3.000 55.000  59.000      Wisconsin   
3128.000  50.000   2.000     3.000 55.000  63.000      Wisconsin   
3133.000  50.000   2.000     3.000 55.000  73.000      Wisconsin   
3137.000  50.000   2.000     3.000 55.000  79.000      Wisconsin   
3141.000  50.000   2.000     3.000 55.000  87.000      Wisconsin   
3148.000  50.000   2.000     3.000 55.000 101.000      Wisconsin   
3150.000  50.000   2.000     3.000 55.000 105.000      Wisconsin   
3156.000  50.000   2.000     3.000 55.000 117.000      Wisconsin   
3161.000  50.000   2.000     3.000 55.000 127.000      Wisconsin  

### Normalizing Data

<p>Now let's start with the initial Z-scores. Here's the formula we'll use to normalize all the data so we can compare apples to apples: z = (x – μ) / σ.</p>

In [41]:
df['zFAMKIDSCHANGE'] = (df.CHANGE_IN_FAM_WITH_KIDS - df.CHANGE_IN_FAM_WITH_KIDS.mean())/ df.CHANGE_IN_FAM_WITH_KIDS.std()

In [42]:
df['zPERCENT_WITH_KIDS'] = (df.PERCENT_WITH_KIDS - df.PERCENT_WITH_KIDS.mean())/ df.PERCENT_WITH_KIDS.std()

In [43]:
df['zBIRTHS_PER_CAPITA'] = (df.BIRTHS_PER_CAPITA - df.BIRTHS_PER_CAPITA.mean())/ df.BIRTHS_PER_CAPITA.std()

In [44]:
df['zAVG_NETMIG'] = (df.AVG_NETMIG - df.AVG_NETMIG.mean())/ df.AVG_NETMIG.std()

In [45]:
df['zPOP_CHANGE'] = (df.POP_CHANGE - df.POP_CHANGE.mean())/ df.POP_CHANGE.std()

In [46]:
df['zINCOME_AS_PERCENT_OF_STATE'] = (df.INCOME_AS_PERCENT_OF_STATE - df.INCOME_AS_PERCENT_OF_STATE.mean())/ df.INCOME_AS_PERCENT_OF_STATE.std()

In [47]:
df['zPERCENTAGE_IN_POVERTY'] = -1*((df.PERCENTAGE_IN_POVERTY - df.PERCENTAGE_IN_POVERTY.mean())/ df.PERCENTAGE_IN_POVERTY.std())

In [48]:
df['zPERCENT_W_HEALTH_INSURANCE'] = (df.PERCENT_W_HEALTH_INSURANCE - df.PERCENT_W_HEALTH_INSURANCE.mean())/ df.PERCENT_W_HEALTH_INSURANCE.std()

In [49]:
df['zUNEMPLOYMENT'] = -1*((df.UNEMPLOYMENT - df.UNEMPLOYMENT.mean())/ df.UNEMPLOYMENT.std())

In [50]:
df['zBACHELORS_DEGREE'] = (df.BACHELORS_DEGREE - df.BACHELORS_DEGREE.mean())/ df.BACHELORS_DEGREE.std()

In [51]:
df['zMORTGAGE_INCOME_RATIO'] = (df.MORTGAGE_INCOME_RATIO - df.MORTGAGE_INCOME_RATIO.mean())/ df.MORTGAGE_INCOME_RATIO.std()

In [52]:
df['zRENT_INCOME_RATIO'] = (df.RENT_INCOME_RATIO - df.RENT_INCOME_RATIO.mean())/ df.RENT_INCOME_RATIO.std()

In [53]:
df['zHEALTHCARE_INCOME_RATIO'] = (df.HEALTHCARE_INCOME_RATIO - df.HEALTHCARE_INCOME_RATIO.mean())/ df.HEALTHCARE_INCOME_RATIO.std()

In [54]:
df['zDOCTORS'] = (df.DOCTORS - df.DOCTORS.mean())/ df.DOCTORS.std()

In [55]:
df['zEXERCISE_ACCESS'] = (df.EXERCISE_ACCESS - df.EXERCISE_ACCESS.mean())/ df.EXERCISE_ACCESS.std()

In [56]:
df['zFOOD_ENVIRONMENT_INDEX'] = (df.FOOD_ENVIRONMENT_INDEX - df.FOOD_ENVIRONMENT_INDEX.mean())/ df.FOOD_ENVIRONMENT_INDEX.std()

In [57]:
df['zCIVIC_ASSOCIATIONS'] = (df.CIVIC_ASSOCIATIONS - df.CIVIC_ASSOCIATIONS.mean())/ df.CIVIC_ASSOCIATIONS.std()

In [58]:
df['zCRIME_RATE'] = -1*((df.CRIME_RATE - df.CRIME_RATE.mean())/ df.CRIME_RATE.std())

In [59]:
df['zDAILY_POLLUTION'] = -1*((df.DAILY_POLLUTION - df.DAILY_POLLUTION.mean())/ df.DAILY_POLLUTION.std())

In [61]:
df['zCHILD_MORTALITY_RATE'] = -1*((df.CHILD_MORTALITY_RATE - df.CHILD_MORTALITY_RATE.mean())/ df.CHILD_MORTALITY_RATE.std())

In [62]:
df['zCHILDREN_UNINSURED'] = -1*((df.CHILDREN_UNINSURED - df.CHILDREN_UNINSURED.mean())/ df.CHILDREN_UNINSURED.std())

In [63]:
df['zDISCONNECTED_YOUTH'] = -1*((df.DISCONNECTED_YOUTH - df.DISCONNECTED_YOUTH.mean())/ df.DISCONNECTED_YOUTH.std())

In [64]:
df['zHOUSING_PROBLEMS'] = -1*((df.HOUSING_PROBLEMS - df.HOUSING_PROBLEMS.mean())/ df.HOUSING_PROBLEMS.std())

### Ranking Counties
<p>Now that we have normalized scores for all of our 23 of our variables. You'll notice, we multiplied some of those by -1, because families want less pollution, crime, etc.</p>

In [68]:
df['zSCORE_TOTAL'] = (df['zPERCENT_WITH_KIDS'] + df['zBIRTHS_PER_CAPITA'] + df['zFAMKIDSCHANGE'] + df['zAVG_NETMIG'] + df['zPOP_CHANGE']  + df['zINCOME_AS_PERCENT_OF_STATE'] + df['zPERCENTAGE_IN_POVERTY'] + df['zPERCENT_W_HEALTH_INSURANCE'] + df['zUNEMPLOYMENT'] + df['zBACHELORS_DEGREE'] + df['zMORTGAGE_INCOME_RATIO'] + df['zRENT_INCOME_RATIO'] + df['zHEALTHCARE_INCOME_RATIO'] + df['zDOCTORS'] + df['zEXERCISE_ACCESS'] + df['zFOOD_ENVIRONMENT_INDEX'] + df['zCIVIC_ASSOCIATIONS'] + df['zCRIME_RATE'] + df['zDAILY_POLLUTION'] + df['zCHILD_MORTALITY_RATE'] + df['zCHILDREN_UNINSURED'] + df['zDISCONNECTED_YOUTH'] + df['zHOUSING_PROBLEMS'])/23

In [69]:
df_sorted = df.sort_values('zSCORE_TOTAL', ascending=False)

In [70]:
df_sorted.head(50)

SUMLEV  REGION  DIVISION  STATE  COUNTY          STNAME  \
2919.000  50.000   3.000     5.000 51.000 107.000        Virginia   
268.000   50.000   4.000     8.000  8.000  35.000        Colorado   
2564.000  50.000   3.000     6.000 47.000 187.000       Tennessee   
740.000   50.000   2.000     3.000 18.000  57.000         Indiana   
2099.000  50.000   2.000     3.000 39.000  41.000            Ohio   
455.000   50.000   3.000     5.000 13.000 117.000         Georgia   
2895.000  50.000   3.000     5.000 51.000  59.000        Virginia   
1226.000  50.000   3.000     5.000 24.000  27.000        Maryland   
1407.000  50.000   2.000     4.000 27.000 139.000       Minnesota   
2873.000  50.000   3.000     5.000 51.000  13.000        Virginia   
1890.000  50.000   1.000     2.000 36.000  59.000        New York   
950.000   50.000   2.000     4.000 20.000  91.000          Kansas   
1228.000  50.000   3.000     5.000 24.000  31.000        Maryland   
1758.000  50.000   2.000     4.000 31.000 153.000        Nebraska   
2609.000  50.000   3.000     7.000 48.000  85.000           Texas   
1247.000  50.000   1.000     1.000 25.000  17.000   Massachusetts   
2939.000  50.000   3.000     5.000 51.000 153.000        Virginia   
1419.000  50.000   2.000     4.000 27.000 163.000       Minnesota   
1249.000  50.000   1.000     1.000 25.000  21.000   Massachusetts   
1392.000  50.000   2.000     4.000 27.000 109.000       Minnesota   
2827.000  50.000   4.000     8.000 49.000  11.000            Utah   
2645.000  50.000   3.000     7.000 48.000 157.000           Texas   
233.000   50.000   4.000     9.000  6.000  85.000      California   
2952.000  50.000   3.000     5.000 51.000 179.000        Virginia   
2015.000  50.000   3.000     5.000 37.000 183.000  North Carolina   
1423.000  50.000   2.000     4.000 27.000 171.000       Minnesota   
1818.000  50.000   1.000     2.000 34.000  27.000      New Jersey   
2812.000  50.000   3.000     7.000 48.000 491.000           Texas   
2962.000  50.000   3.000     5.000 51.000 510.000        Virginia   
3017.000  50.000   4.000     9.000 53.000  33.000      Washington   
3163.000  50.000   2.000     3.000 55.000 131.000       Wisconsin   
655.000   50.000   2.000     3.000 17.000  93.000        Illinois   
3164.000  50.000   2.000     3.000 55.000 133.000       Wisconsin   
2627.000  50.000   3.000     7.000 48.000 121.000           Texas   
231.000   50.000   4.000     9.000  6.000  81.000      California   
1822.000  50.000   1.000     2.000 34.000  35.000      New Jersey   
2328.000  50.000   1.000     2.000 42.000  91.000    Pennsylvania   
1356.000  50.000   2.000     4.000 27.000  37.000       Minnesota   
1600.000  50.000   2.000     4.000 29.000 183.000        Missouri   
2161.000  50.000   2.000     3.000 39.000 165.000            Ohio   
1814.000  50.000   1.000     2.000 34.000  19.000      New Jersey   
118.000   50.000   3.000     7.000  5.000   7.000        Arkansas   
2846.000  50.000   4.000     8.000 49.000  49.000            Utah   
1215.000  50.000   3.000     5.000 24.000   3.000        Maryland   
1906.000  50.000   1.000     2.000 36.000  91.000        New York   
384.000   50.000   3.000     5.000 12.000 109.000         Florida   
881.000   50.000   2.000     4.000 19.000 153.000            Iowa   
743.000   50.000   2.000     3.000 18.000  63.000         Indiana   
2908.000  50.000   3.000     5.000 51.000  85.000        Virginia   
3109.000  50.000   2.000     3.000 55.000  25.000       Wisconsin   

                                      CTY  HEALTHCARE_COST  \
2919.000         Loudoun County, Virginia         8272.000   
268.000          Douglas County, Colorado         9521.000   
2564.000     Williamson County, Tennessee         8545.000   
740.000          Hamilton County, Indiana         9303.000   
2099.000            Delaware County, Ohio         9350.000   
455.000           Forsyth County, Georgia        10233.000   
2895.000         Fairfax County, Virginia         7799.0

In [71]:
df_sorted.to_csv("/Users/alexmahadevan/Code/raise_a_family/county data sorted.csv")

<p>Just glancing at the top few counties, you can see affordability kind of got lost in there, even though we did include healthcare, rent and mortgage costs in the analysis.</p>
<p>To do the next part of the analysis, I'm only picking out the top 200 counties.</p> 
<p>Then I'll hard-code the school grades from [Niche](https://www.niche.com/k12/search/best-school-districts/), and re-run the analysis putting more weight on affordability. That should produce our final list.</p>